catalog/security/catalog-security-logging/src/main/java/org/codice/ddf/catalog/security/logging/SecurityLoggingPlugin.java

```diff
@@ -13,85 +13,260 @@
+
+    @Override
+    public ResourceResponse process(ResourceResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        String addtional = "Return resource: " + input.getResource()
+                .getName();
+        logOperation(CatalogOperationType.RESOURCE_RESPONSE, input, addtional);
         return input;
     }
 
-    private void logOperation(CatalogOperationType operationType, Request request) {
-        String user;
-        try {
-            Subject subject = SecurityUtils.getSubject();
-            user = SubjectUtils.getName(subject, NO_USER);
-        } catch (Exception e) {
-            user = NO_USER;
+    @Override
+    public QueryRequest process(Source source, QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        String additional = "";
+        Query query = input.getQuery();
+        if (query instanceof QueryImpl) {
+            additional = ((QueryImpl) query).getFilter()
+                    .toString();
         }
+        logOperation(CatalogOperationType.QUERY_REQUEST, input, additional);
+        return input;
+    }
+
+    @Override
+    public CreateStorageRequest process(CreateStorageRequest input)
+            throws PluginExecutionException {
+        String additional = "Creating products: " + input.getContentItems()
+                .stream()
+                .map(ContentItem::getMetacard)
+                .map(Metacard::getId)
+                .collect(Collectors.joining(","));
+        logOperation(CatalogOperationType.INGEST_REQUEST, input, additional);
+        return input;
+    }
+
+    @Override
+    public UpdateStorageRequest process(UpdateStorageRequest input)
+            throws PluginExecutionException {
+        String additional = "Updating products: " + input.getContentItems()
+                .stream()
+                .map(ContentItem::getMetacard)
+                .map(Metacard::getId)
+                .collect(Collectors.joining(","));
+        logOperation(CatalogOperationType.UPDATE_REQUEST, input, additional);
+        return input;
+    }
+
+    @Override
+    public CreateStorageResponse process(CreateStorageResponse input)
+            throws PluginExecutionException {
+        String additional = "Created product: " + input.getCreatedContentItems()
+                .stream()
+                .map(ContentItem::getMetacard)
+                .map(Metacard::getId)
+                .collect(Collectors.joining(","));
+        logOperation(CatalogOperationType.INGEST_RESPONSE, input, additional);
+        return input;
+    }
+
+    @Override
+    public UpdateStorageResponse process(UpdateStorageResponse input)
+            throws PluginExecutionException {
+        String additional = "Updated products: " + input.getUpdatedContentItems()
+                .stream()
+                .map(ContentItem::getMetacard)
+                .map(Metacard::getId)
+                .collect(Collectors.joining(","));
```

Comment:
There is an overloaded version of the `joining` method that takes a prefix and suffix. I believe you would get the same output with:
```java
String additional = input.getUpdatedContentItems()
    .stream()
    .map(ContentItem::getMetacard)
    .map(Metacard::getId)
    .collect(Collectors.joining(",", "Updated products: ", "");
```